File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/bindinstance/simple/GodRebrandingComponent.java
package com.github.klee0kai.test_ext.inject.di.bindinstance.simple;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.component.ExtendOf;
import com.github.klee0kai.test.di.bindinstance.simple.GodWorkspaceComponent;

@Component
public interface GodRebrandingComponent extends GodWorkspaceComponent {

    @Override
    SunSystemRebrandingModule sunSystem();

    @ExtendOf
    void extendComponent(GodWorkspaceComponent parent);

}

File: tests_ext/src/main/java/com/github/klee0kai/test_ext/inject/di/bindinstance/simple/GodRebrandingComponent.kt
package com.github.klee0kai.test_ext.inject.di.bindinstance.simple

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.ExtendOf
import com.github.klee0kai.test.di.bindinstance.simple.GodWorkspaceComponent

@Component
interface GodRebrandingComponent : GodWorkspaceComponent {

    override fun sunSystem(): SunSystemRebrandingModule

    @ExtendOf
    fun extendComponent(parent: GodWorkspaceComponent)

}
